db/migrations/versions/028_validate_check_constraints.py

```diff
@@ -0,0 +1,31 @@
+"""validate check constraints
+
+Revision ID: 028_validate_check_constraints
+Revises: 027_start_date_before_end
+Create Date: 2024-04-05 08:03:43.847606
+
+"""
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "028_validate_check_constraints"
+down_revision = "027_start_date_before_end"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    for table in [
+        "funding",
+        "programme_funding_management",
+        "project_progress",
+        "outcome_data",
+        "output_data",
+        "submission_dim",
+    ]:
+        op.execute(f"ALTER TABLE {table} VALIDATE CONSTRAINT ck_{table}_start_before_end")
```

Comment:
Have you deployed this branch to the dev environment to see what happens when we run this? Dev is very close to Prod in terms of what's in the database so would be a good reflection of whether this will validate successfully or not

Reply:
Just did - and it blew up as you perhaps suspected ;)

Will chat with Gideon to work out how we find+fix the dodgy data

Reply:
The main problem with putting  this into 2 migrations is that if on Prod the 3nd one fails, then we're no better off than if they were just one and it all failed, as basically our migrations will be out of sync with code. But if all in one migration, then it will all roll back atomically if there is an issue